plothist/plotters.py

```diff
@@ -469,107 +321,237 @@ def plot_error_hist(hist, ax, **kwargs):
 
-    fig, (ax_comparison, ax_ratio) = plt.subplots(
-        2, gridspec_kw={"height_ratios": [4, 1]}
-    )
+    if fig is None and ax_main is None and ax_comparison is None:
+        fig, (ax_main, ax_comparison) = create_comparison_figure()
+    elif (fig is not None or ax_main is not None or ax_comparison is not None) and (
+        fig is None or ax_main is None or ax_comparison is None
+    ):
+        raise ValueError(
+            "Need to provid fig, ax_main and ax_comparison (or None of them)."
+        )
 
     xlim = (hist_1.axes[0].edges[0], hist_1.axes[0].edges[-1])
 
-    plot_hist(hist_1, ax=ax_comparison, label=x1_label, histtype="step")
-    plot_hist(hist_2, ax=ax_comparison, label=x2_label, histtype="step")
-    ax_comparison.set_xlim(xlim)
-    ax_comparison.set_ylabel(ylabel)
-    ax_comparison.tick_params(axis="x", labelbottom="off")
-    ax_comparison.legend(framealpha=0.5)
+    plot_hist(hist_1, ax=ax_main, label=x1_label, histtype="step")
+    plot_hist(hist_2, ax=ax_main, label=x2_label, histtype="step")
+    ax_main.set_xlim(xlim)
+    ax_main.set_ylabel(ylabel)
+    ax_main.tick_params(axis="x", labelbottom="off")
+    ax_main.legend(framealpha=0.5)
+    _ = ax_main.xaxis.set_ticklabels([])
+    fig.subplots_adjust(hspace=0.125)
+
+    plot_comparison(
+        hist_1,
+        hist_2,
+        ax_comparison,
+        xlabel=xlabel,
+        x1_label=x1_label,
+        x2_label=x2_label,
+        comparison=comparison,
+        comparison_ylim=comparison_ylim,
+        ratio_uncertainty=ratio_uncertainty,
+    )
+
+    if save_as is not None:
+        fig.savefig(save_as, bbox_inches="tight")
+
+    return fig, ax_main, ax_comparison
+
+
+def plot_comparison(
+    hist_1,
+    hist_2,
+    ax,
+    xlabel="x1",
+    x1_label="x1",
+    x2_label="x2",
+    comparison="ratio",
+    comparison_ylim=None,
+    ratio_uncertainty="uncorrelated",
+):
+    """
+    Plot the comparison between two histograms.
+
+    Parameters
+    ----------
+    hist_1 : boost_histogram.Histogram
+        The first histogram for comparison.
+    hist_2 : boost_histogram.Histogram
+        The second histogram for comparison.
+    ax : matplotlib.axes.Axes
+        The axes to plot the comparison.
+    xlabel : str, optional
+        The label for the x-axis. Default is "x1".
+    x1_label : str, optional
+        The label for the first histogram. Default is "x1".
+    x2_label : str, optional
+        The label for the second histogram. Default is "x2".
+    comparison : str, optional
+        The type of comparison to plot ("ratio" or "pull"). Default is "ratio".
+    comparison_ylim : tuple or None, optional
+        The y-axis limits for the comparison plot. Default is None.
+    ratio_uncertainty : str, optional
+        How to treat the uncertainties of the histograms when comparison = "ratio" ("uncorrelated" for simple comparison, "split" for scaling and split hist_1 and hist_2 uncertainties)
```

Comment:
Let's add `"efficiency"` at the next PR.

Reply:
Let's also say explicitly that this argument has no effect if `comparison` is not `ratio`